erts/preloaded/src/erlang.erl

```diff
@@ -5219,6 +5221,67 @@ Example:
 processes() ->
     erlang:nif_error(undefined).
 
+%% The process iterator is a 2-tuple, consisting of an index to the process
+%% table and a list of process identifiers that existed when the last scan of
+%% the process table took place. The index is the starting place for the next
+%% scan of the process table.
+-opaque processes_iter_ref() :: {integer(), [pid()]}.
+
+%% processes_iterator/0
+-doc """
+Returns a processes iterator that can be used in
+[`processes_next/1`](`processes_next/1`).
+""".
+-doc #{ group => processes }.
+-spec processes_iterator() -> processes_iter_ref().
+processes_iterator() ->
+    {0, []}.
+
+%% processes_next/1
+-doc """
+Returns a 2-tuple, consisting of one process identifier and a new processes
+iterator. If the process iterator has run out of processes in the process table,
+`none` will be returned.
+
+Example:
+
+```erlang
+> I0 = erlang:processes_iterator(), ok.
+ok
+> {Pid1, I1} = erlang:processes_next(I0), Pid1.
+<0.0.0>,
+> {Pid2, I2} = erlang:processes_next(I1), Pid2.
+<0.1.0>
+```
+
+> #### Note {: .info }
+>
+> This BIF has less consistency guarantee than [`processes/0`](`processes/0`).
+> Process identifiers returned from consecutive calls of this BIF may not be a
+> consistent snapshot of all elements existing in the table during any of the
+> calls. The process identifier of a process that is alive before
+> `processes_iterator/0` is called and continues to be alive until
+> `processes_next/1` returns `none` is guaranteed to be part of the result
+> returned from one of the calls to `processes_next/1`.
```

Comment:
```suggestion
> returned from one of the calls to `processes_next/1`.
>
> The two major benefits of using the `processes_iterator()`/`processes_next()` BIFs instead of using
> the `processes()` BIF are that they scale better since no locking is needed, and you do not risk
> getting a huge list allocated on the heap if there are a huge amount of processes alive in the
> system.
```

It could be good to list the benefits of using these BIFs instead of the `processes()` BIF. Above I put a suggestion that will be inside the Note-box, perhaps it should be outside of the Note-box, though. You decide.